soh/soh/Enhancements/gameplaystats.cpp

```diff
@@ -135,196 +325,196 @@ void DrawStatsTracker(bool& open) {
-                DisplayStatIfNonZero("B:                  ", gSaveContext.sohStats.count[COUNT_BUTTON_PRESSES_B]);
-                DisplayStatIfNonZero("L:                  ", gSaveContext.sohStats.count[COUNT_BUTTON_PRESSES_L]);
-                DisplayStatIfNonZero("R:                  ", gSaveContext.sohStats.count[COUNT_BUTTON_PRESSES_R]);
-                DisplayStatIfNonZero("Z:                  ", gSaveContext.sohStats.count[COUNT_BUTTON_PRESSES_Z]);
-                DisplayStatIfNonZero("C-Up:               ", gSaveContext.sohStats.count[COUNT_BUTTON_PRESSES_CUP]);
-                DisplayStatIfNonZero("C-Right:            ", gSaveContext.sohStats.count[COUNT_BUTTON_PRESSES_CRIGHT]);
-                DisplayStatIfNonZero("C-Down:             ", gSaveContext.sohStats.count[COUNT_BUTTON_PRESSES_CDOWN]);
-                DisplayStatIfNonZero("C-Left:             ", gSaveContext.sohStats.count[COUNT_BUTTON_PRESSES_CLEFT]);
-                DisplayStatIfNonZero("D-Up:               ", gSaveContext.sohStats.count[COUNT_BUTTON_PRESSES_DUP]);
-                DisplayStatIfNonZero("D-Right:            ", gSaveContext.sohStats.count[COUNT_BUTTON_PRESSES_DRIGHT]);
-                DisplayStatIfNonZero("D-Down:             ", gSaveContext.sohStats.count[COUNT_BUTTON_PRESSES_DDOWN]);
-                DisplayStatIfNonZero("D-Left:             ", gSaveContext.sohStats.count[COUNT_BUTTON_PRESSES_DLEFT]);
-                DisplayStatIfNonZero("Start:              ", gSaveContext.sohStats.count[COUNT_BUTTON_PRESSES_START]);
-
-                ImGui::NewLine();
-                ImGui::TreePop();
+        if (ImGui::BeginTabItem("Breakdown")) {
+            UIWidgets::PaddedEnhancementCheckbox("Room Breakdown", "gGameplayStatRoomBreakdown");
+            ImGui::SameLine();
+            UIWidgets::InsertHelpHoverText("Allows a more in-depth perspective of time spent in a certain map.");
+            if (gPlayState == NULL) {
+                ImGui::Text("Waiting for file load...");
+            } else {
+                for (int i = 0; i < gSaveContext.sohStats.tsIdx; i++) {
+                    TimestampInfo tsInfo = sceneTimestampDisplay[i];
+                    bool canShow = CVarGetInteger("gGameplayStatRoomBreakdown", 0) ? true : !(tsInfo.isRoom);
+                    if (tsInfo.time > 0 && strnlen(tsInfo.name, 40) > 1 && canShow) {
+                        DisplayTimeHHMMSS(tsInfo.time, tsInfo.name, tsInfo.color);
+                    }
+                }
+                std::string toPass;
+                if (CVarGetInteger("gGameplayStatRoomBreakdown", 0) && gSaveContext.sohStats.sceneNum != SCENE_KAKUSIANA) {
+                    toPass = fmt::format("{:s} Room {:d}", ResolveSceneID(gSaveContext.sohStats.sceneNum, gSaveContext.sohStats.roomNum), gSaveContext.sohStats.roomNum);
+                } else {
+                    toPass = fmt::format("{:s}", ResolveSceneID(gSaveContext.sohStats.sceneNum, gSaveContext.sohStats.roomNum));
```

Comment:
do we need the format here? can this just be
```suggestion
                    toPass = ResolveSceneID(gSaveContext.sohStats.sceneNum, gSaveContext.sohStats.roomNum);
```